src/arithmetic/canonical.cc

```diff
@@ -746,6 +746,22 @@ Expr Simplify(Expr a, Map<Var, Range> vrange) {
   // and it only appears as the top op in an expression. So we strip it
   // first and send the sub-expressions to the simplifier.
   if (const Reduce* r = a.as<Reduce>()) {
+    // If axis is empty, we can remove the reduce op completely.
```

Comment:
Is there any reason not to call `SimplifyCombiner` inside `Simplify`? I'm not sure if it's necessary to export a new API with similar functionality as `Simplify`.

Reply:
SimplifyCombiner does two things:
- Simplifies parts of the combiner. This should be ok to do by default in Simplify.
- Removes unused components of the combiner assuming that only one component is really needed (with the index `value_index`). This transformation may be unsafe when this assumption is false (like when we have an operation returning multiple values), so I would expect something to break if this is enabled by default, but I haven't checked.

So, I think I will try calling SimplifyCombiner with different parameters from Simplify and see if something breaks.

Reply:
Well, I found an argument against it. If one wants to simplify a multi-output reduction operation represented as a `ComputeOpNode` by calling `Simplify` on the components of its `body`, the resulting array of bodies may contain incoherent `Reduce` expressions (and `VerifyComputeOp` will fail). However it is more of a theoretical concern because:
- Nobody does that as it seems
- It is incorrect to call `Simplify` this way even with the current implementation of Simplify because `VerifyComputeOp` compares fields of reductions by reference.